test/notifications/display_test.dart

```diff
@@ -495,6 +534,112 @@ void main() {
         expectedTitle: eg.selfUser.fullName,
         expectedTagComponent: 'dm:${message.allRecipientIds.join(",")}');
     })));
+
+    test('remove: smoke', () => runWithHttpClient(() => awaitFakeAsync((async) async {
+      await init();
+      final message = eg.streamMessage();
+      final data = messageFcmMessage(message);
+      final expectedGroupKey = '${data.realmUri}|${data.userId}';
+
+      check(testBinding.androidNotificationHost.activeNotifications).isEmpty();
+      await receiveFcmMessage(async, data);
+      check(testBinding.androidNotificationHost.activeNotifications).deepEquals(<Condition<Object?>>[
+        conditionActiveNotif(data, 'stream:${message.streamId}:${message.topic}'),
+        conditionSummaryActiveNotif(expectedGroupKey),
+      ]);
+      testBinding.firebaseMessaging.onMessage.add(
+        RemoteMessage(data: removeFcmMessage([message]).toJson()));
+      async.flushMicrotasks();
+      check(testBinding.androidNotificationHost.activeNotifications).isEmpty();
+
+      check(testBinding.androidNotificationHost.activeNotifications).isEmpty();
+      await receiveFcmMessage(async, data);
+      check(testBinding.androidNotificationHost.activeNotifications).deepEquals(<Condition<Object?>>[
+        conditionActiveNotif(data, 'stream:${message.streamId}:${message.topic}'),
+        conditionSummaryActiveNotif(expectedGroupKey),
+      ]);
+      testBinding.firebaseMessaging.onBackgroundMessage.add(
+        RemoteMessage(data: removeFcmMessage([message]).toJson()));
+      async.flushMicrotasks();
+      check(testBinding.androidNotificationHost.activeNotifications).isEmpty();
+    })));
+
+    test('remove: clears conversation only if the removal event is for the last message', () => runWithHttpClient(() => awaitFakeAsync((async) async {
+      await init();
+      final stream = eg.stream();
+      const topicA = 'Topic A';
+      final message1 = eg.streamMessage(stream: stream, topic: topicA);
+      final data1 = messageFcmMessage(message1, streamName: stream.name);
+      final message2 = eg.streamMessage(stream: stream, topic: topicA);
+      final data2 = messageFcmMessage(message2, streamName: stream.name);
+      final message3 = eg.streamMessage(stream: stream, topic: topicA);
+      final data3 = messageFcmMessage(message3, streamName: stream.name);
+      final expectedGroupKey = '${data1.realmUri}|${data1.userId}';
+
+      check(testBinding.androidNotificationHost.activeNotifications).isEmpty();
+
+      await receiveFcmMessage(async, data1);
+      await receiveFcmMessage(async, data2);
+      await receiveFcmMessage(async, data3);
+
+      check(testBinding.androidNotificationHost.activeNotifications).deepEquals(<Condition<Object?>>[
```

Comment:
nit:

```suggestion
      await receiveFcmMessage(async, data1);
      await receiveFcmMessage(async, data2);
      await receiveFcmMessage(async, data3);
      check(testBinding.androidNotificationHost.activeNotifications).deepEquals(<Condition<Object?>>[
```

Generally when a test has code like "make this change, this change, this change; then check the state of the world", I like to group those into one stanza. Then a blank line separates the checks from the next round of making changes.

That way the stanzas, or the blank lines separating them, serve to group together those change/check pairs.